### PR TITLE
Correct the restrictions on data-* attribute names

### LINKED ISSUE #2751: Issue with "data-*": ('semicolon' does not match presented unicode value)
MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*

#### What information was incorrect, unhelpful, or incomplete?
"the name must not contain any semicolon (U+003A);"

'semicolon' does not match unicode value (it matches the 'colon' value)

#### Specific section or headline?
data-* top level content

#### What did you expect to see?
Either 
"the name must not contain any colon (U+003A);"
OR
"the name must not contain any semicolon (U+003B);"

#### Did you test this? If so, how?
"\u003a" == ':'
// => true
"\u003b" == ';' 
// => true

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/html/global_attributes/data-_star_`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/data-*
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/global_attributes/data-_star_/index.html
* Last commit: https://github.com/mdn/content/commit/f6f11d2a249520f01bad93d7c4a5e5cdf6ed914a
* Document last modified: 2021-02-19T19:33:30.000Z

</details>

### PR DESCRIPTION
**Pull request by CreaTorAlexander**
Fixes #2751

### CODE REVIEW

File: files/en-us/web/html/global_attributes/data-_star_/index.html [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 <ul>
  <li>the name must not start with <code>xml</code>, whatever case is used for these letters;</li>
- <li>the name must not contain any semicolon (<code>U+003A</code>);</li>
+ <li>the name must not contain any semicolon (<code>U+003B</code>);</li>
```
Comment (sideshowbarker) [inline files/en-us/web/html/global_attributes/data-_star_/index.html:22]:
```suggestion
 <li>The name must not contain any colon characters (<code>:</code>).</li>
```
When submitting a patch like this one, you need to read the relevant spec requirements, and make sure the change aligns with the spec. In this case, the relevant spec requirements are here:

* https://html.spec.whatwg.org/multipage/dom.html#custom-data-attribute
* https://html.spec.whatwg.org/multipage/infrastructure.html#xml-compatible

File: files/en-us/web/html/global_attributes/data-_star_/index.html [hunk 2]
```diff
@@ -19,7 +19,7 @@
...
  <li>the name must not contain capital letters.</li>
```
Comment (sideshowbarker) [inline files/en-us/web/html/global_attributes/data-_star_/index.html:23]:
```suggestion
 <li>The name must not contain any capital letters.</li>
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

File: files/en-us/web/html/global_attributes/data-_star_/index.html [hunk 3]
```diff
@@ -19,7 +19,7 @@
 
 <ul>
  <li>the name must not start with <code>xml</code>, whatever case is used for these letters;</li>
```
Comment (sideshowbarker) [inline files/en-us/web/html/global_attributes/data-_star_/index.html:21]:
```suggestion
 <li>The name must not start with <code>xml</code> (case-insensitive).</li>
```

**Review by sideshowbarker — CHANGES_REQUESTED** [review]

File: files/en-us/web/html/global_attributes/data-_star_/index.html (hunk 1 above)
Comment (CreaTorAlexander) [reply files/en-us/web/html/global_attributes/data-_star_/index.html:22]: True, thanks for the patch. I assumed it has just the wrong Unicode.

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**